### PR TITLE
add ng-apimock for dev mocking

### PR DESCRIPTION
**Pull request by groninge01**
- add backend mocking

### CODE REVIEW

**Comment by garethfuller** [comment]
@groninge01 closing for now, feel free to reopen if you want to keep working on it.

File: package.json [hunk 1]
```diff
@@ -54,6 +56,7 @@
     "@vercel/speed-insights": "^1.0.1",
     "bignumber.js": "^9.1.2",
     "chakra-react-select": "^4.7.6",
+    "cors": "^2.8.5",
```
Comment (agualis) [inline package.json:59]: If this is only required by `ng-apimock` feature we should move it to `devDependencies`

File: test/ng-apimock/mocks/data/getPool.json [hunk 2]
```diff
@@ -0,0 +1,247 @@
+{
```
Comment (agualis) [inline test/ng-apimock/mocks/data/getPool.json:1]: Better rename to `getPool.mock.json` to match other similar json mock files ([getPool.mock.json](https://github.com/balancer/frontend-v3/pull/769/files#diff-67659822893fc2a6cc737fcad54bf178f91454a43e04e409be71581e984a7aa9))

File: test/ng-apimock/mocks/data/getPoolSnapshots.json [hunk 3]
```diff
@@ -0,0 +1,285 @@
+{
```
Comment (agualis) [inline test/ng-apimock/mocks/data/getPoolSnapshots.json:1]: Rename to `getPoolSnapshots.mock.json`

**Review by agualis — CHANGES_REQUESTED** [review]
Small changes requested.

Looks goods to me if you find it useful. As I said, I would like to try to have the same testing feature with MSW to remove this `ng-apiMock` dependency but we can take a look at that in the future.

File: test/ng-apimock/mocks/data/getPool.json (hunk 2 above)
Comment (groninge01) [reply test/ng-apimock/mocks/data/getPool.json:1]: this is the (response) data for the `getPool` mock

File: test/ng-apimock/mocks/data/getPoolSnapshots.json (hunk 3 above)
Comment (groninge01) [reply test/ng-apimock/mocks/data/getPoolSnapshots.json:1]: this is the (response) data for the `getPoolSnapshots` mock

**Comment by groninge01** [comment]
> Small changes requested.
> 
> Looks goods to me if you find it useful. As I said, I would like to try to have the same testing feature with MSW to remove this `ng-apiMock` dependency but we can take a look at that in the future.

Let's wait for that then. If I need mocks before that's ready I'll just checkout this branch.

**Pull request closed without merging**